src/com/palmergames/bukkit/towny/TownyUniverse.java

```diff
@@ -501,21 +505,19 @@ public TownBlock getTownBlock(WorldCoord worldCoord) throws NotRegisteredExcepti
 			throw new NotRegisteredException();
 	}
 
-	/**
-	 * Get Universe-wide ConcurrentHashMap of WorldCoords and their TownBlocks.
-	 * Populated at load time from townblocks folder's files.
-	 * 
-	 * 
-	 * @return townblocks hashmap read from townblock files.
-	 */	
-	public Map<WorldCoord, TownBlock> getTownBlocks() {
-		return townBlocks;
+	@Override
+	@NotNull
+	public @UnmodifiableView Collection<TownBlock> getTownBlocks() {
+		return townBlocks.values();
 	}
 	
-	public void addTownBlock(TownBlock townBlock) {
-		if (hasTownBlock(townBlock.getWorldCoord()))
-			return;
+	@Override
+	public boolean addTownBlock(@NotNull TownBlock townBlock) {
+		if (hasTownBlock(townBlock.getWorldCoord())) {
```

Comment:
Store `townBlock.getWorldCoord()` to prevent duplicate object creation.

Alternatively, can also just make the entire function one line by doing `return townBlocks.putIfAbsent(townBlock.getWorldCoord(), townBlock) == null;`

`putIfAbsent` will return null if there was no previous value associated with the key.